google-cloud-core/lib/google/cloud.rb

```diff
@@ -156,9 +156,12 @@ def self.loaded_files
     # @private
     #
     def self.auto_load_files
-      return Gem.find_latest_files "google-cloud-*.rb" if Gem.respond_to? :find_latest_files
-      # Ruby 2.0 does not have Gem.find_latest_files
-      Gem.find_files "google-cloud-*.rb"
+      Gem.loaded_specs.values.filter_map do |spec|
+        next unless spec.name.start_with?("google-cloud-")
+        path = File.join(spec.lib_dirs_glob, spec.name + ".rb")
```

Comment:
My understanding is, `spec.lib_dirs_glob` can return glob syntax (i.e. including commas etc) in cases where there are multiple source paths. That would cause this to fail. Maybe we can iterate over `spec.require_paths` instead? (Note that `require_paths` is relative to `spec.full_gem_path`, and can also be `nil`, where we should default to `"lib"`.)